server/server.go

```diff
@@ -26,6 +33,21 @@ func (cs *cribbageServer) Serve() {
 		})
 	})
 
+	// Group: auth
+	authGroup := router.Group(`/auth`)
+	{
+		authGroup.POST(`/login`, cs.ginPostLogin)
```

Comment:
It seems like this login path should just be handled by middleware. Can't we have middleware handle _all_ of our groups -- validating that the user making the request is logged in and allowed?

Reply:
The middleware can be used on a group (seen in the `private` group example) to protect a set of routes. I assume we'll want some public routes and some private. I'm not sure the middleware should be defining any endpoints itself?

Reply:
Can we have every route (except "home") go through the auth middleware? And if a user isn't authed, then send them to the homepage?

Reply:
Sure, it just depends which routes we need protected. If we think it's all of them except the home page (I suppose that's probably what we want), then this would be fine